static/app/components/forms/formField/index.tsx

```diff
@@ -160,162 +170,140 @@ type PassthroughProps = Omit<
   | 'defaultValue'
 >;
 
-class FormField extends Component<FormFieldProps> {
-  static defaultProps = {
-    hideErrorMessage: false,
-    flexibleControlStateSize: false,
-  };
-
-  componentDidMount() {
-    // Tell model about this field's props
-    this.getModel().setFieldDescriptor(this.props.name, this.props);
-  }
-
-  componentWillUnmount() {
-    this.getModel().removeField(this.props.name);
-  }
-
-  static contextType = FormContext;
-
-  getError() {
-    return this.getModel().getError(this.props.name);
-  }
-
-  getId() {
-    return sanitizeQuerySelector(this.props.name);
-  }
-
-  getModel(): FormModel {
-    return this.context.form !== undefined
-      ? this.context.form
-      : new MockModel(this.props);
-  }
-
-  input: HTMLElement | null = null;
-
-  /**
-   * Attempts to autofocus input field if field's name is in url hash.
-   *
-   * The ref must be forwarded for this to work.
-   */
-  handleInputMount = (node: HTMLElement | null) => {
-    if (node && !this.input) {
-      // TODO(mark) Clean this up. FormContext could include the location
-      const hash = window.location?.hash;
+function FormField(props: FormFieldProps) {
+  const initialProps = useRef(props);
 
-      if (!hash) {
-        return;
-      }
+  const {name, onBlur, onChange, onKeyDown} = props;
 
-      if (hash !== `#${this.props.name}`) {
-        return;
-      }
+  const context = useContext(FormContext);
+  const inputRef = useRef<HTMLElement>();
 
-      // Not all form fields have this (e.g. Select fields)
-      if (typeof node.focus === 'function') {
-        node.focus();
-      }
-    }
+  const [model] = useState<FormModel>(
+    // XXX: MockModel doesn't fully implement the FormModel interface
+    () => context.form ?? (new MockModel(props) as any)
+  );
 
-    this.input = node;
-  };
+  // Register field within the model
+  useEffect(() => {
+    model.setFieldDescriptor(name, initialProps.current);
+    return () => model.removeField(name);
+  }, [model, name]);
```

Comment:
The old logic was only run on mount, couldn't this end-up passing stale props to `model.setFieldDescription()` should `name` change?

Reply:
Yeah it could, but that's basically the same behavior as run on mount right?

Reply:
As long as the name and model never change :smile:

Reply:
theoretically, the name wouldn't change after the component was mounted, right?